engine/execution/ingestion/engine.go

```diff
@@ -1139,8 +1139,11 @@ func (e *Engine) saveExecutionResults(
 		span, _ := e.tracer.StartSpanFromContext(childCtx, trace.EXESaveTransactionEvents)
 		defer span.Finish()
 
-		if len(events) > 0 {
-			err = e.events.Store(blockID, events)
+		// store events 1K in each batch
+		chunkSize := uint(1000)
```

Comment:
Do you know if the transaction limit in Badger is metered by DB operation or by data size? If we have really large events will this still work?

Reply:
seems to be a limit based on the ratio of the table size https://discuss.dgraph.io/t/discussion-badgerdb-should-offer-arbitrarily-sized-atomic-transactions/8736 :-| 
we might have to change the "chunkify" to act based on the event size.

Reply:
Further down that discussion they point to https://github.com/dgraph-io/badger/blob/ef28ef36b5923f12ffe3a1702bdfa6b479db6637/batch.go#L37-L42

That might be a good fit for this.

Reply:
but considering the event byte size limit that we are going to put in place, I think we can come up with some limits for tx based on the counts.

Reply:
The issue with have with the batch writer is we do look up for IDs I guess before writing, we might change it completely to use the batch writer and do upsert style of storage for events

Reply:
> The issue with have with the batch writer is we do look up for IDs I guess before writing

I think to use the batch writer you anyway need to re-write that `operation` level function to call `batch.Set` rather than `tx.Set`. But you're right it would change the semantics.

Reply:
@jordanschalm do you there is a value in merging this PR or we should create an issue to properly address this issue by size ?

Reply:
I'm good to merge this. It seems like it will only make the problem better, even if it doesn't fully handle all cases. But we should make an issue to implement a solution that uses `BatchWriter` or that guarantees we won't hit the limits implicitly with event/tx size/count limits (I prefer `BatchWriter` though).